Exercise #3 - Js Objects. Part1.

Humarik opened the pull request:
This pull request implements tasks 2-6

Humarik committed:
Add task-02

The showObjElem is function that writes property name and
value of each property of object in console.

Humarik committed:
Add task-03

The сheckObj is function that accepts string and object
and after this returns true or false
if object has property name same as string

Humarik committed:
Add task-04

The checkObj is function that accepts string and object
and after this checks the object
if object does't have property name same as in string then
new property will be added with name as in string and
value "new" will be added there

Humarik committed:
Add task-05

The copyObject is function that accepts object and
after this makes copy of this object and returns new copied object

Humarik committed:
Add task-06

The deepCopyObj is function that makes deep copy obj with help recursion

Humarik committed:
Fix tasks-05,06

Task-05 has been changed because I used assign() instead of native js
Task-06 has been changed because I made mistake in cycle
and pushed elements of array without recursion

Humarik committed:
Fix task-06

Function deepCopyObj has been changed because I noticed mistake
in my code but last test passed successfully.
Any way I needed to fix my homework. I found article where says that
I can use cycle 'for in' for array too. After this my code's became less

aneelia commented on src/ex3_js-objects-part1/task-03.js:
```diff
@@ -0,0 +1,7 @@
+function checkObj(str, obj) {
+    if(str in obj){
```
в одну строку можно записать всю функцию

aneelia commented on src/ex3_js-objects-part1/task-04.js:
```diff
@@ -0,0 +1,11 @@
+function checkObj(str, obj) {
+    const thatObj = obj;
+    switch (str in thatObj) {
```
здесь switch не уместен

Humarik committed:
Fix task-03,04

Funtion has been changed and rewritten in one string in task-03.
Linter required to add extra variable for object and also
add brackets when we assign to variable 'new' in task-04
